File: java/42587.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<Integer> queue = new LinkedList<>();
        
        // 큐에 넣어주기.
        for (int tmp : priorities) {
            queue.add(tmp);
        }
        
        int queueSize = queue.size();
        while (queueSize > 0) {            
            int max = Collections.max(queue);
            int tmp = queue.peek();
            
            if (max == tmp) {
                // 우선순위가 가장 높으므로, 뽑아서 실행해도 된다.
                queue.remove();
                queueSize --;
                answer ++; // 업무를 수행했으므로.
                
                if (location == 0) {
                    return answer;
                }
                
                location --;
            } else {
                // 우선순위가 안맞으므로, 다시 큐에 넣는다.
                queue.remove();
                queue.add(tmp);
                
                if (location == 0) {
                    location = queueSize - 1; // 큐 가장 뒤에 우리가 찾는 업무가 있다는 것
                } else {
                    // location은 계속해서 추적해주기
                    location --;
                }
            }
        }
        
        return answer;
    }
}

File: python/42587.py
from collections import deque

def solution(priorities, location):
    orgLen = len(priorities)
    priorities = deque(priorities) #큐를 구현하기 위한 것 (시간절약)
    outProcess = 0 # 현재 실행된 프로세스 개수
    
    while len(priorities) > 1:
        tmp = priorities.popleft()
        
        if tmp >= max(priorities):
            # 큐에서 꺼낸 값이 가장 크다면 실행.
            if location == 0:
                # 내가 찾는 값이 나온 것
                return outProcess + 1
            else:
                # 내가 찾는 값이 아니면, 넣진 말고
                outProcess = outProcess + 1        
        else:
            priorities.append(tmp) #다시 큐에 넣기
            
            if location == 0:
                location = len(priorities) - 1
                continue
        
        location = location-1
    
    # 여기까지 왔다는 것은 이미 마지막까지 돌았다는 것.
    return orgLen